### ISSUE TITLE
mongo was stopped after deployment

### ISSUE THREAD

**Issue by marrobi** [opencb/opencga#1151]
● mongod.service - High-performance, schema-free document-oriented database
   Loaded: loaded (/lib/systemd/system/mongod.service; enabled; vendor preset: enabled)
   Active: inactive (dead)
     Docs: https://docs.mongodb.org/manual

**Comment by lawrencegripper** [comment]
So this looks like it's a timing issue with the mongo setup scripts based on the extract from the cloud-init logs below. 

```
...
The partition table has been altered.
Calling ioctl() to re-read partition table.
Syncing disks.
mke2fs 1.42.13 (17-May-2015)
The file /dev/sdc1 does not exist and no size was specified.
Cloud-init v. 18.4-0ubuntu1~16.04.2 running 'modules:final' at Tue, 19 Feb 2019 11:07:14 +0000. Up 39.97 seconds.
...
```

By the time I got to the box `/dev/sdc1` existed 

```
opencgaadmin@mongo-vm0:~$ ls /dev/sd*
/dev/sda  /dev/sda1  /dev/sdb  /dev/sdb1  /dev/sdc  /dev/sdc1
```

Going to carry on digging

cc: @bart-jansen and @marrobi

**Comment by lawrencegripper** [comment]
From the looks of the logs there is maybe a slim change that sometimes `sfdisk` exits before finishing or there is a very close race. I've added the following while look to ensure the disk partition exists before attempting to format it. 

```bash
#partitions primary linux partition on new disk
    echo "Partitioning disk"
    echo 'type=83' | sfdisk ${1}
    while [ ! -f /tmp/list.txt ]
    do
        echo "Waiting for drive to be partitioned"
        sleep 2
    done
    echo "Formatting Partition"
    mkfs -t ext4 ${1}1
```

Going to test out now.

**Comment by lawrencegripper** [comment]
So turns out you can't do the `-f` syntax on drive partitions so I've moved to `blkid` which prints the information about the block device or exists if it doesn't exist with `exit 2` then wrapped this in the `until` syntax

```
until blkid ${1}1
    do
        echo "Waiting for drive to be partitioned"
        sleep 2
    done
```

**Comment by lawrencegripper** [comment]
This looks to have worked on my last deployment, due to the timing nature of this one I'm going to keep it open and if we don't see a repeat in other deployments lets close it off. 

Fix is in #1164

**Comment by marrobi** [comment]
Not seen it again, done lots of mongo deploys so closing.
